okhttp-tests/src/test/java/okhttp3/ConnectionReuseTest.java

```diff
@@ -301,14 +301,22 @@
    * https://github.com/square/okhttp/issues/2409
    */
   @Test public void connectionsAreNotReusedIfNetworkInterceptorInterferes() throws Exception {
-    client = client.newBuilder().addNetworkInterceptor(new Interceptor() {
-      @Override public Response intercept(Chain chain) throws IOException {
-        Response response = chain.proceed(chain.request());
-        return response.newBuilder()
-            .body(ResponseBody.create(null, "unrelated response body!"))
-            .build();
-      }
-    }).build();
+    client = client.newBuilder()
+        // Since this test knowingly leaks a connection, avoid using the default shared connection
+        // pool, which should remain clean for subsequent tests.
+        .connectionPool(new ConnectionPool())
```

Comment:
Is it worth cleaning up this pool also?

Reply:
It does makes sense, but I'm not aware of a way to do this using ConnectionPool's public methods. Let me know if you can think of a way.

Reply:
@yschimke, any idea on how to do this?

Reply:
I'm not aware of a real-world problem that calls for adding such a public API, but I'm going to leave this outside the scope of this PR.

Reply:
Mainly being a good Android citizen.  It's a legitimate thing to want to be able to shutdown an App cleanly, close connections etc.  You may even want to do something like this on backgrounding an app to save battery etc.

Reply:
Should we open an issue for this?

Reply:
@swankjesse has always pushed back on these types of close methods etc, I think generally because the good usage patterns of client instances often reuse connection pools and dispatchers, and the close APIs can corrupt that pattern.

Reply:
There is a way to shut down an idle HTTP client. But it's cumbersome.

I'm reluctant to make it easier because I don't want people to think of their OkHttp clients as things they need to stress about managing. And I really don't want people doing the obvious-looking thing and creating a new client for every call.